include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/GenerateRobolectricPreviewTestsExtension.kt

```diff
@@ -0,0 +1,236 @@
+  variant: Variant,
+  scanPackages: ListProperty<String>
+) {
+  assert(scanPackages.get().orEmpty().isNotEmpty()) {
+    "Please set roborazzi.generateRobolectricPreviewTests.packages in the generatePreviewTests extension or set roborazzi.generateRobolectricPreviewTests.enable = false." +
+      "See https://github.com/sergio-sastre/ComposablePreviewScanner?tab=readme-ov-file#how-to-use for more information."
+  }
+
+  val generateTestsTask = project.tasks.register(
+    "generate${variant.name.capitalize()}PreviewScreenshotTests",
+    GeneratePreviewScreenshotTestsTask::class.java
+  ) {
+    // It seems that this directory path is overridden by addGeneratedSourceDirectory.
+    // The generated tests will be located in build/JAVA/generate[VariantName]PreviewScreenshotTests.
+    it.outputDir.set(project.layout.buildDirectory.dir("generated/roborazzi/preview-screenshot"))
+    it.scanPackageTrees.set(scanPackages)
+  }
+  // We need to use sources.java here; otherwise, the generate task will not be executed.
+  // https://stackoverflow.com/a/76870110/4339442
+  variant.unitTest?.sources?.java?.addGeneratedSourceDirectory(
+    generateTestsTask,
+    GeneratePreviewScreenshotTestsTask::outputDir
+  )
+}
+
+private fun verifyMavenRepository(project: Project) {
+  // Check if the jitpack repository is added.
+  val hasJitpackRepo = project.repositories.any {
+    it is MavenArtifactRepository &&
+      it.url.toString().contains("https://jitpack.io")
+  }
+  if (!hasJitpackRepo) {
+    error(
+      "Roborazzi: Please add the following 'maven' repository to the 'repositories' block in the 'build.gradle' file.\n" +
+        "build.gradle: maven { url 'https://jitpack.io' }\n" +
+        "build.gradle.kts: maven { url = uri(\"https://jitpack.io\") }\n" +
+        "This is necessary to download the ComposablePreviewScanner."
+    )
+  }
+}
+
+private fun verifyTestConfig(
+  testTaskProvider: TaskCollection<Test>,
+  androidExtension: TestedExtension,
+  logger: Logger
+) {
+  if (!androidExtension.testOptions.unitTests.isIncludeAndroidResources) {
+    logger.warn(
+      "Roborazzi: Please set 'android.testOptions.unitTests.isIncludeAndroidResources = true' in the 'build.gradle' file. " +
+        "This is advisable to avoid issues with ActivityNotFoundException."
+    )
+  }
+  testTaskProvider.configureEach { testTask ->
+    if (testTask.systemProperties["roborazzi.record.filePathStrategy"] != "relativePathFromRoborazziContextOutputDirectory") {
+      logger.info(
+        "Roborazzi: Please set 'roborazzi.record.filePathStrategy=relativePathFromRoborazziContextOutputDirectory' in the 'gradle.properties' file. " +
+          "This is advisable to avoid unnecessary path manipulation. " +
+          "Please refer to 'https://github.com/takahirom/roborazzi?tab=readme-ov-file#roborazzirecordfilepathstrategy' for more information."
+      )
+    }
+    if (testTask.systemProperties["robolectric.pixelCopyRenderMode"] != "hardware") {
```

Comment:
Rather than supporting older versions, maybe we should warn if using legacy versions? Are there typically reasons not to update robolectric if some new library you introduce uses it?

Reply:
Thanks. I added a little comment for it.
https://github.com/takahirom/roborazzi/pull/416/commits/a7984d09452715d7ffdd18e3b4cc21795d47ee78